PolyEngine/Engine/Src/Entity.hpp

```diff
@@ -16,12 +18,48 @@ namespace Poly {
 		const UniqueID& GetID() const { HEAVY_ASSERTE(EntityID, "Entity was not properly initialized");  return EntityID; }
 		const World* GetWorld() const { HEAVY_ASSERTE(EntityID, "Entity was not properly initialized");  return EntityWorld; }
 
+		//////////////////////////////
+		/// Note: This function should only be accessed by engine's internal logic.
+		/// Checks whether there is a component of a given ID under this Entity's ID.
+		/// @param ID of a component type
+		/// @see GetComponent()
+		/// @see HasComponents()
+		//////////////////////////////
+		bool HasComponent(size_t ID) const;
+		
+		//////////////////////////////
+		/// Note: This function should only be accessed by engine's internal logic.
+		/// Checks whether there are all of the specified components under this Entity's ID.
+		/// @param IDs is a bit field of component IDs to be checked for, see macro BIT()
+		/// @see GetComponent()
+		/// @see HasComponent()
+		//////////////////////////////
+		inline bool HasComponents(unsigned long int IDs) const
+		{
+			return ComponentPosessionFlags.to_ullong() & IDs;
```

Comment:
This method will throw exception in size of the bit set is greater than 64bits. Let's leave it for now and add a comment stating that BitMask class impleentation is required here + a STATIC_ASSERTE for MAX_COMPONENTS_COUNT <= 64.